src/typecheck.c

```diff
@@ -159,7 +160,9 @@ static void do_implicit_cast(
             && !(from->kind == TYPE_SIGNED_INTEGER && to->kind == TYPE_UNSIGNED_INTEGER)
         ) || (
             // Cast from any integer type to double.
-            is_integer_type(from) && to == doubleType
+            is_integer_type(from) && to == doubleType
+        ) || (
+            is_integer_type(from) && to == floatType
```

Comment:
Assuming `is_float_type()` checks for `float` or `double`:
```suggestion
            is_integer_type(from) && is_float_type(to)
```